python_modules/dagster/dagster_tests/daemon_tests/test_monitoring_daemon.py

```diff
@@ -151,42 +176,156 @@ def test_monitor_starting(instance: DagsterInstance, logger: Logger):
+        too_long_run = create_run_for_test(
+            instance,
+            pipeline_name="foo",
+            status=DagsterRunStatus.STARTING,
+            tags={MAX_RUNTIME_TAG: "500"},
+        )
+        okay_run = create_run_for_test(
+            instance,
+            pipeline_name="foo",
+            status=DagsterRunStatus.STARTING,
+            tags={MAX_RUNTIME_TAG: "1000"},
+        )
+        run_no_tag = create_run_for_test(
+            instance, pipeline_name="foo", status=DagsterRunStatus.STARTING
+        )
+    started_time = initial.add(seconds=1)
+    with pendulum.test(started_time):
+        report_started_event(instance, too_long_run, started_time.timestamp())
+        report_started_event(instance, okay_run, started_time.timestamp())
+        report_started_event(instance, run_no_tag, started_time.timestamp())
+
+    too_long_record = instance.get_run_record_by_id(too_long_run.run_id)
+    assert too_long_record is not None
+    assert too_long_record.dagster_run.status == DagsterRunStatus.STARTED
+    assert too_long_record.start_time == started_time.timestamp()
+
+    okay_record = instance.get_run_record_by_id(okay_run.run_id)
+    assert okay_record is not None
+    assert okay_record.dagster_run.status == DagsterRunStatus.STARTED
+    assert okay_record.start_time == started_time.timestamp()
+
+    no_tag_record = instance.get_run_record_by_id(run_no_tag.run_id)
+    assert no_tag_record is not None
+    assert no_tag_record.dagster_run.status == DagsterRunStatus.STARTED
+    assert no_tag_record.start_time == started_time.timestamp()
+
+    workspace = workspace_context.create_request_context()
+    run_launcher = cast(TestRunLauncher, instance.run_launcher)
+
+    eval_time = started_time.add(seconds=501)
+    with pendulum.test(eval_time):
+        # run_no_tag has no maximum run tag set, so no termination event should be
+        # triggered.
+        monitor_started_run(instance, workspace, no_tag_record, logger)
+        run = instance.get_run_by_id(okay_record.dagster_run.run_id)
+        assert run
+        assert not run_launcher.termination_calls
+
+        # Not enough time has elapsed for okay_run to hit its maximum runtime, so no
+        # termination event should be triggered
+        monitor_started_run(instance, workspace, okay_record, logger)
+        run = instance.get_run_by_id(okay_record.dagster_run.run_id)
+        assert run
+        assert not run_launcher.termination_calls
+
+        # Enough runtime has elapsed for too_long_run to hit its maximum runtime so a
+        # termination event should be triggered.
+        monitor_started_run(instance, workspace, too_long_record, logger)
+        run = instance.get_run_by_id(too_long_record.dagster_run.run_id)
+        assert run
+        assert len(run_launcher.termination_calls) == 1
+
+
+def test_long_running_termination_failure(
+    instance: DagsterInstance, workspace_context: WorkspaceProcessContext, logger: Logger
+):
```

Comment:
similar-ish test case that verifies that it still terminates even if the terminate doesn't happen cleanly would be nice (as per suggestion above)

Reply:
by terminate cleanly - you mean forcing that it enters failure state (check new tests I think they handle this case as you would expect now)